chef/cookbooks/provisioner/recipes/setup_base_images.rb

```diff
@@ -70,98 +58,132 @@
+    end
+    break
+  end
 end
 
 # UEFI config
 use_elilo = true
+%w(x86_64 ia32).each do |arch|
+  # Make it easy to totally disable/enable architectures
+  next unless discovery_arches.include? arch
 
-if node[:platform] != "suse"
-  bash "Install elilo as UEFI netboot loader" do
-    code <<EOC
-cd #{uefi_dir}
-tar xzf '#{tftproot}/files/elilo-3.14-all.tar.gz'
-mv elilo-3.14-x86_64.efi bootx64.efi
-mv elilo-3.14-ia32.efi bootia32.efi
-mv elilo-3.14-ia64.efi bootia64.efi
-rm elilo*.efi elilo*.tar.gz || :
-EOC
-    not_if "test -f '#{uefi_dir}/bootx64.efi'"
+  uefi_dir = "#{discovery_dir}/#{arch}/#{uefi_subdir}"
+
+  short_arch = arch
+  if arch == "x86_64"
+    short_arch = "x64"
   end
-else
-  if node["platform_version"].to_f < 12.0
-    package "elilo"
 
-    bash "Install bootx64.efi" do
-      code "cp /usr/lib64/efi/elilo.efi #{uefi_dir}/bootx64.efi"
-      not_if "cmp /usr/lib64/efi/elilo.efi #{uefi_dir}/bootx64.efi"
+  directory uefi_dir do
+    recursive true
+    mode 0755
+    owner "root"
+    group "root"
+    action :create
+  end
+
+  if node[:platform] != "suse"
+    bash "Install elilo as UEFI netboot loader" do
+      code <<EOC
+  cd #{uefi_dir}
+  tar xf '#{tftproot}/files/elilo-3.14-all.tar.gz' boot#{short_arch}.efi
+EOC
+      not_if "test -f '#{uefi_dir}/boot#{short_arch}.efi'"
     end
   else
-    # we use grub2; steps taken from
-    # https://github.com/openSUSE/kiwi/wiki/Setup-PXE-boot-with-EFI-using-grub2
-    use_elilo = false
+    if node["platform_version"].to_f < 12.0
+      package "elilo"
 
-    package "grub2-x86_64-efi"
+      bash "Install boot#{short_arch}.efi" do
+        code "cp /usr/lib64/efi/elilo.efi #{uefi_dir}/boot#{short_arch}.efi"
+        not_if "cmp /usr/lib64/efi/elilo.efi #{uefi_dir}/boot#{short_arch}.efi"
+      end
+    else
+      # we use grub2; steps taken from
+      # https://github.com/openSUSE/kiwi/wiki/Setup-PXE-boot-with-EFI-using-grub2
+      use_elilo = false
 
-    template "#{uefi_dir}/grub.conf" do
+      package "grub2-#{arch}-efi"
+
+      # grub.cfg has to be in boot/grub/ subdirectory
+      directory "#{uefi_dir}/default/boot/grub" do
+        recursive true
+        mode 0755
+        owner "root"
+        group "root"
+        action :create
+      end
+
+      template "#{uefi_dir}/default/boot/grub/grub.cfg" do
+        mode 0644
+        owner "root"
+        group "root"
+        source "grub.conf.erb"
+        variables(append_line: "#{append_line} crowbar.state=discovery",
+                  install_name: "Crowbar Discovery Image",
+                  admin_ip: admin_ip,
+                  initrd: "#{arch}/initrd0.img",
+                  kernel: "#{arch}/vmlinuz0")
+      end
+
+      bash "Build UEFI netboot loader with grub" do
+        cwd "#{uefi_dir}/default"
+        code "grub2-mkstandalone -d /usr/lib/grub2/#{arch}-efi/ -O #{arch}-efi --fonts=\"unicode\" -o #{uefi_dir}/boot#{short_arch}.efi boot/grub/grub.cfg"
```

Comment:
Metrics/LineLength: Line is too long. [155/150] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)